src/declarations/stencil-private.ts

```diff
@@ -1398,19 +1404,34 @@ export interface MinifyJsResult {
   };
 }
 
+/**
+ * A mapping from a TypeScript or JavaScript source file path on disk, to a Stencil {@link Module}.
+ *
+ * It is advised that the key (path) be normalized before storing/retrieving the `Module` to avoid unnecessary lookup
+ * failures.
+ */
 export type ModuleMap = Map<string, Module>;
 
 /**
  * Stencil's Intermediate Representation (IR) of a module, bundling together
  * various pieces of information like the classes declared within it, the path
  * to the original source file, HTML tag names defined in the file, and so on.
  *
- * Note that this gets serialized/parsed as JSON and therefore cannot be a
+ * Note that this gets serialized/parsed as JSON and therefore cannot contain a
  * `Map` or a `Set`.
  */
 export interface Module {
   cmps: ComponentCompilerMeta[];
+  /**
+   * A collection of modules that a component will need. The modules in this list must have import statements generated
+   * in order for the component to function.
+   */
   coreRuntimeApis: string[];
+  /**
+   * A collection of modules that a component will need for a specific output target. The modules in this list must
+   * have import statements generated in order for the component to function, but only for a specific output target.
+   */
+  outputTargetCoreRuntimeApis: Record<string, string[]>;
```

Comment:
just realized this, we might want to type this `Record<OutputTarget['type'], string[]>` here just to prevent any typos in the future

Reply:
makes sense!